packages/contracts/src/core/permission/PermissionManager.sol

```diff
@@ -255,12 +255,16 @@ abstract contract PermissionManager is Initializable {
         address _where,
         address _who,
         bytes32 _permissionId,
-        IPermissionCondition _condition
+        PermissionConditionBase _condition
     ) internal virtual {
         if (!address(_condition).isContract()) {
             revert ConditionInvalid(_condition);
         }
 
+        if (!_condition.supportsInterface(type(IPermissionCondition).interfaceId)) {
+            revert ConditionInvalid(_condition);
+        }
+
         if (_where == ANY_ADDR && _who == ANY_ADDR) {
```

Comment:
do we still need the first check "isContract" ?
if yes, why not have 2 custom error one for isContract and one for supportedInterface ?

Reply:
> do we still need the first check "isContract" ?

We don't strictly need it, but it is much better error handling.
Otherwise, it just reverts without error if you provide an address instead of a contract.

> if yes, why not have 2 custom error one for isContract and one for supportedInterface ?
Not a strong preference here. I can add the second custom error